fetcher/fetcher.go

```diff
@@ -106,18 +96,35 @@ func New(
 		opt(f)
 	}
 
-	// Override transport idle connection settings
-	//
-	// See this conversation around why `.Clone()` is used here:
-	// https://github.com/golang/go/issues/26013
-	customTransport := http.DefaultTransport.(*http.Transport).Clone()
-	customTransport.IdleConnTimeout = DefaultIdleConnTimeout
-	customTransport.MaxIdleConns = f.maxConnections
-	customTransport.MaxIdleConnsPerHost = f.maxConnections
+	if f.rosettaClient == nil {
+		// Override transport idle connection settings
+		//
+		// See this conversation around why `.Clone()` is used here:
+		// https://github.com/golang/go/issues/26013
+		defaultTransport := http.DefaultTransport.(*http.Transport).Clone()
+		defaultTransport.IdleConnTimeout = DefaultIdleConnTimeout
+		defaultTransport.MaxIdleConns = f.maxConnections
+		defaultTransport.MaxIdleConnsPerHost = DefaultMaxConnections
+		defaultHTTPClient := &http.Client{
+			Timeout:   DefaultHTTPTimeout,
+			Transport: defaultTransport,
+		}
+
+		// Create default fetcher
+		clientCfg := client.NewConfiguration(
+			serverAddress,
+			DefaultUserAgent,
+			defaultHTTPClient,
+		)
+		client := client.NewAPIClient(clientCfg)
+		f.rosettaClient = client
```

Comment:
nit: seems like we can consolidate them into (but it's trivial)
```suggestion
		f.rosettaClient = client.NewAPIClient(clientCfg)
```